app/models.py

```diff
@@ -869,5 +869,30 @@ class StalenessSchema(MarshmallowSchema):
     immutable_time_to_stale_warning = fields.Integer(validate=marshmallow_validate.Range(min=1, max=2147483647))
     immutable_time_to_delete = fields.Integer(validate=marshmallow_validate.Range(min=1, max=2147483647))
 
+    @validates_schema
+    def validate_staleness(self, data, **kwargs):
+        if data["conventional_time_to_stale"] >= data["conventional_time_to_stale_warning"]:
+            raise MarshmallowValidationError(
+                "conventional_time_to_stale must be lower than conventional_time_to_stale_warning"
+            )
+        elif data["conventional_time_to_stale"] >= data["conventional_time_to_delete"]:
+            raise MarshmallowValidationError(
+                "conventional_time_to_stale must be lower than conventional_time_to_delete"
+            )
+        elif data["conventional_time_to_stale_warning"] >= data["conventional_time_to_delete"]:
+            raise MarshmallowValidationError(
+                "conventional_time_to_stale_warning must be lower than conventional_time_to_delete"
+            )
+        elif data["immutable_time_to_stale"] >= data["immutable_time_to_stale_warning"]:
+            raise MarshmallowValidationError(
+                "immutable_time_to_stale must be lower than immutable_time_to_stale_warning"
+            )
+        elif data["immutable_time_to_stale"] >= data["immutable_time_to_delete"]:
+            raise MarshmallowValidationError("immutable_time_to_stale must be lower than immutable_time_to_delete")
+        elif data["immutable_time_to_stale_warning"] >= data["immutable_time_to_delete"]:
+            raise MarshmallowValidationError(
+                "immutable_time_to_stale_warning must be lower than immutable_time_to_delete"
+            )
+
```

Comment:
In order to reduce the repetition here, I suggest something like this:
```suggestion
        staleness_fields = ["time_to_stale", "time_to_stale_warning", "time_to_delete"]
        for host_type in ("conventional", "immutable",):
            for i in range(len(staleness_fields) - 1): # For all but the last field
                for j in range(i+1, len(staleness_fields)): # For all fields after that field
                    if data[(field_1 := f"{host_type}_{staleness_fields[i]}")] >= data[(field_2 := f"{host_type}_{staleness_fields[j]}")]:
                        raise MarshmallowValidationError(
                            f"{field_1} must be lower than {field_2}"
                        )
```

`field_1` and `field_2` can be instantiated on separate lines if L878 is too hard to read.